### ISSUE TITLE
Remove test dir from PyPi package?

### ISSUE THREAD

**Issue by pierrepo** [pierrepo/PBxplore#121]
Dear all,
I've experienced very slow uploads of new releases of PBxplore to PyPi with my local internet connection. The final package is about 16 Mo on PyPi:
https://pypi.python.org/pypi/pbxplore/1.3

In order to lighten the package size, could we consider removing the test directory (test scripts + test data) from the PyPi distribution system?
I believe this test directory is made for development not production. In addition, the CI system ensures that PBxplore is 'fine' on most Python distributions.
If needed, a user can 'test' PBxplore with data provided in the demo directory, as explained in the documentation.
This will roughly reduce the size of the package by half.

@HubLot & @jbarnoud what do you think about this?

**Comment by jbarnoud** [comment]
I like that streamline users can test that everything work with their 
install. So I vote to keep the tests on pypi somehow. It does not have 
to be in the main pbxplore package, though. The way MDAnalysis does it 
is to have a separate package for the tests; we could do that. Note that 
it would require to tweak PBdata a bit (and have it work only if the 
tests are installed).

**Comment by pierrepo** [comment]
Good point!
However, I believe you are more an 'advanced' user than a 'normal' one.
Personally, I never tried to test packages installed through PyPi (maybe I should) and I'm pretty sure a lambda user won't either ;-)

**Comment by HubLot** [comment]
I like also having tests within the package... It's more convenient for user who wants to contribute.
IMO, I don't really like the way of MDAnalysis separate things. I think it's understandable for a large package with many different tests on many different input files (like MDA) but for PBxplore, it's overkill and add too much complexity.

One reasonably solution could be reduce the size of the test & demo directories. XTC can be shrinked to a few Ko (instead of 13Mo right know in `test_data`). We could also use smaller PDB (190 chains in `demo/beta3_IEGF12.pdb` for 8Mo).
We can easily decrease by 2-3 the size of the package with these tricks

**Comment by jbarnoud** [comment]
Also, some files could be compressed.

**Comment by pierrepo** [comment]
Yes but I think you mix two different types of users:
- normal users who just expect PBxplore to work out of the box and who want to install PBxplore as smoothly and quickly as possible
- advanced users who might like to improve PBxplore, the GitHub repo is then made for them.

That why I believe the test dir is useless in the PyPi package ;-)

**Comment by jbarnoud** [comment]
I would add a category: normal users that will not touch the code, but 
want to make sure everything is OK in their install as they will rely on 
the program for their paper.

**Comment by pierrepo** [comment]
Agree but that is why we use a CI system and also why PyPi allows requirement.txt to have every dependence right.

Most of my published results are produced by software such Gromacs, PyMOL, Python, R, Numpy, Pandas... and I never performed a single test on these software, assuming they were working correctly (buy maybe I was wrong).

**Comment by jbarnoud** [comment]
Travis tells that it should be fine. It does not say that it works fine 
on your machine. You may use unexpected version of some dependencies for 
instance.

**Comment by pierrepo** [comment]
You're right about Travis.
But to test that PBxplore works correctly, you need to:
- install PBxplore,
- go to the path where PBxplore has been installed (tricky to find for a normal user),
- find the correct command to test PBxplore (again, not easy for a normal user).
Apart @jbarnoud and @HubLot , I do not know anybody that tests his packages installed through PyPi.
So I proposed, we remove tests from the PyPi package but add "advanced users" section in the documentation that explains how to get the software from GitHub and run the test scripts.

**Comment by HubLot** [comment]
Okay fair enough.

**Comment by pierrepo** [comment]
Thanks

![](http://i.giphy.com/MyFuWu0Pfq4BW.gif)

**Comment by pierrepo** [comment]
Hi guys,

I found in this thread that is was possible to have tests available in the package namespace (as Numpy does) :
http://stackoverflow.com/questions/32390705/should-i-include-tests-and-pyc-files-when-building-package-with-setuptools

It would be great -- and a good way to keep tests in the PyPi package -- to do such a thing:

```Python
import pbxplore
pbxplore.test()
```
or
```Python
import pbxplore
pbxplore.test_regression()
pbxplore.test_functions()
```
with of course test_regression.py and test_functions.py still available to be run standalone as we do in the development process.

Do you think we could implement this? @jbarnoud do you like this?

**Comment by pierrepo** [comment]
@HubLot could you also shrink a bit test data as you proposed?

**Comment by HubLot** [comment]
fix with #122
